### PR TITLE
feat(sqllab): Show sql in the current result

### PR DESCRIPTION
**Pull request by justinpark**
### SUMMARY
Following #24329, SQLLab can run a portion of the sql editor without highlighted. 
Like other 3rd party sql editor, it will be useful to display the current running sql in the result panel to identify the actual executed query.

Screenshot_2023-07-24_at_3_46_19_PM

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

After:

https://github.com/apache/superset/assets/1392866/eda3ac07-3173-469f-9991-0a9587981917

Before:

https://github.com/apache/superset/assets/1392866/158f8a2d-0d8a-473c-982f-1e89c4b18414

### TESTING INSTRUCTIONS
Go to sqllab
Run a query and check the result panel

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by michael-s-molina** [comment]
I'll leave the design review for @kasiazjc but I think we can truncate the query (add ...) to indicate that we're only showing part of the query.

**Comment by kasiazjc** [comment]
> I'll leave the design review for @kasiazjc but I think we can truncate the query (add ...) to indicate that we're only showing part of the query.

Thanks @michael-s-molina and thanks @justinpark for the PR! 

I have a few suggestions: 
- in other places in the app we actually have a number of rows or the right side of the table and filters/title etc on the left side; I think it would make sense to switch the order in here too, because the sql line acts as a title. Below a screenshot for reference (from explore) 
image
- The sql field could have a max width and if the text (first line) doesn't fill that space, the sql input could shrink a little bit
- if there is another line, more text that is longer than the max width of the container (as Michael said) there should be a truncation rule with "..." - this is a default truncation behavior we use across the app 
- and the last one - this is more of a question - I'm wondering if instead of showing a pop up with the sql query, would it be more useful to scroll and highlight the query in the sql query area at the top? Not sure if that would be even possible, but curious to hear your thoughts @michael-s-molina and @justinpark

**Comment by michael-s-molina** [comment]
> * in other places in the app we actually have a number of rows or the right side of the table and filters/title etc on the left side; I think it would make sense to switch the order in here too, because the sql line acts as a title. Below a screenshot for reference (from explore)

Wow, that's a great suggestion. @justinpark Could we also display the number of rows using the Label (pill) component?

> * and the last one - this is more of a question - I'm wondering if instead of showing a pop up with the sql query, would it be more useful to scroll and highlight the query in the sql query area at the top? Not sure if that would be even possible, but curious to hear your thoughts @michael-s-molina and @justinpark

@justinpark If this is possible it would be an awesome UX!

**Comment by justinpark** [comment]
> and the last one - this is more of a question - I'm wondering if instead of showing a pop up with the sql query, would it be more useful to scroll and highlight the query in the sql query area at the top? 

@kasiazjc this is a good idea but it might be broken if a user modify the sql content which is not same from the ran query.

**Comment by justinpark** [comment]
> Could we also display the number of rows using the Label (pill) component?

Sure. I'll update accordingly

**Comment by kasiazjc** [comment]
> > and the last one - this is more of a question - I'm wondering if instead of showing a pop up with the sql query, would it be more useful to scroll and highlight the query in the sql query area at the top?
> 
> @kasiazjc this is a good idea but it might be broken if a user modify the sql content which is not same from the ran query.

True, so I think for now let's keep the modal. Thanks!

**Comment by justinpark** [comment]
> I have a few suggestions:

> in other places in the app we actually have a number of rows or the right side of the table and filters/title etc on the left side; I think it would make sense to switch the order in here too, because the sql line acts as a title. Below a screenshot for reference (from explore)
> image
> The sql field could have a max width and if the text (first line) doesn't fill that space, the sql input could shrink a little bit
> 
> if there is another line, more text that is longer than the max width of the container (as Michael said) there should be a truncation rule with "..." - this is a default truncation behavior we use across the app

@kasiazjc I updated the layout as you suggested. I also replaces the existing limit message by the tooltip (+ exclamation indicator when exists). Feel free to give any feedback at your convenience.


https://github.com/apache/superset/assets/1392866/80a73d05-0534-48ea-a7f5-2691578275bc

**Comment by justinpark** [comment]
@kasiazjc circle back related to the update

**Comment by justinpark** [comment]
@kasiazjc @michael-s-molina could you help a review again?

File: superset-frontend/src/SqlLab/components/ResultSet/index.tsx
Comment (justinpark) [reply superset-frontend/src/SqlLab/components/ResultSet/index.tsx]: @michael-s-molina updated by theme variables.

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Comment by justinpark** [comment]
@kasiazjc I will remain open until next Tuesday and will merge it if you don't have any concerns.

**Comment by jinghua-qa** [comment]
I filed this cosmetic issue for the yellow banner of the query limit, it looks like the yellow banner is now on the edge of the query filed.
https://github.com/apache/superset/issues/25288

**Pull request merged**
